lib/ecto/migration.ex

```diff
@@ -528,7 +527,11 @@ defmodule Ecto.Migration do
   defp do_create(table, command) do
     columns =
       if table.primary_key do
-        [{:add, :id, :bigserial, primary_key: true}]
+        opts = Runner.repo_config(:migration_primary_key, [])
+        opts = Keyword.put(opts, :primary_key, true)
+        {name, opts} = Keyword.pop(opts, :name, :id)
+        {type, opts} = Keyword.pop(opts, :type, :bigserial)
```

Comment:
these 4 lines are duplicated (see the top of this diff) but not sure if it's worth extracting out.

Reply:
I refactored this in 78f876b.